packages/components/src/placeholder/index.js

```diff
@@ -27,13 +27,19 @@ function Placeholder( {
 	...additionalProps
 } ) {
 	const [ resizeListener, { width } ] = useResizeAware();
-	const classes = classnames(
-		'components-placeholder',
-		width >= 320 ? 'is-large' : '',
-		width >= 160 && width < 320 ? 'is-medium' : '',
-		width < 160 ? 'is-small' : '',
-		className
-	);
+
+	// Since `useResizeAware` will report a width of `null` until after the
+	// first render, avoid applying any modifier classes until width is known.
```

Comment:
Alternatively you could manually measure the element sizes and apply them as default values.

```js
const [ resizeListener, { width = elementRef.current.clientWidth } ] = useResizeAware();
```

Reply:
Hi @FezVrasta. If I understand correctly, I don't think that would be possible.

- A ref to a DOM node would only be assigned after the first render. It was my understanding this was also the reason `react-resize-aware` would provide `null` initially.
- The specific syntax wouldn't work, since the defaulting would not take effect if the value of width is `null` (only if it were undefined).

```js
const sizes = { width: null };
const { width = 'somedefault' } = sizes;
console.log( width );
// null
```

Reply:
My bad, you are right, you would need an outer component with its own wrapping markup to do that.